src/context/directory/handlers/clientGrants.ts

```diff
@@ -39,15 +40,45 @@ async function dump(context: DirectoryContext): Promise<void> {
   const grantsFolder = path.join(context.filePath, constants.CLIENTS_GRANTS_DIRECTORY);
   fs.ensureDirSync(grantsFolder);
 
+  if (clientGrants.length === 0) return;
```

Comment:
@sergiught This is the optimization I originally forgot to include. If there are no client grants to dump, we should prevent the process from fetching clients and resource servers.